File: leetcode/algorithms/hash-table/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        result = None;
        compMap = defaultdict(lambda: None);
        
        for i in range(len(nums)):
            comp = target - nums[i];
            compMapResult = compMap[comp];
            if compMapResult!=None:
                result = [compMapResult,i];
            compMap[nums[i]] = i;
        return result;

File: leetcode/algorithms/hash-table/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer,Integer> compMap = new HashMap<Integer,Integer>();
        int[] solArr = {-1,-1}; 
        
        for(int i=0;i<nums.length;++i){
            Integer compResult = compMap.getOrDefault(target - nums[i],null);
            if(compResult!=null){
                solArr = new int[]{i,compResult};
                return solArr;
            }
            compMap.put(nums[i],i);
        }
        
        return solArr;
        
    }
}